### ISSUE TITLE
TypeError exception on array comparison

### ISSUE THREAD

**Issue by febeling** [Starcounter-Jack/JSON-Patch#85]
This test passes with 0.5.4 and fails with 0.5.5
```javascript
    it('jsonpatch array compare', function () {
      var a = ['foo', 'bar']
      var b = ['foo', 'bar', 'baz']

      expect(jsonpatch.compare(b, a)).toEqual([
        { op: 'remove', path: '/2' }
      ])
      expect(jsonpatch.compare(a, b)).toEqual([
        { op: 'add', path: '/2', value: 'baz' }
      ])
    })
```
This is the stacktrace:
```
PhantomJS 1.9.8 (Mac OS X 0.0.0) Service: utility changes to get from updated to original jsonpatch array compare FAILED
	TypeError: '[object Object]' is not a valid argument for 'Function.prototype.apply' (evaluating 'obj.length')
	    at /Users/febeling/work/ehealth/sl-ebola-call-admin/app/bower_components/fast-json-patch/src/json-patch-duplex.js:29
	    at _generate (/Users/febeling/work/ehealth/sl-ebola-call-admin/app/bower_components/fast-json-patch/src/json-patch-duplex.js:405)
	    at compare (/Users/febeling/work/ehealth/sl-ebola-call-admin/app/bower_components/fast-json-patch/src/json-patch-duplex.js:537)
	    at /Users/febeling/work/ehealth/sl-ebola-call-admin/test/spec/services/utility.js:117
```
_Updated: showing tests exercising the library API directly, instead of our thin service layer._

**Comment by warpech** [comment]
@miyconst could you please take a look?

**Comment by miyconst** [comment]
Sure.

**Comment by miyconst** [comment]
@febeling, could you please provide your full testing environment?

And could you try to replace

```js
return Array.apply(null, { length: obj.length }).map(function (e, i, a) { return i.toString(); });
```
at https://github.com/Starcounter-Jack/JSON-Patch/blob/master/src/json-patch-duplex.js#L29

with 

```js
var keys = new Array(obj.length);

for (var i = 0; i < keys.length; i++) {
	keys[i] = i.toString();
}

return keys;
```

And let me know whether it passes.

**Comment by febeling** [comment]
The tests are written with jasmine and run in karma, controlling a phantomjs browser runtime.

I can't reproduce the problem in plain node.

The problem goes away when switching karma to use Chrome as browser runtime.

With the change to loop iteration you propose above it also works with phantomjs.

**Comment by miyconst** [comment]
Should be fixed in [0.5.6](https://github.com/Starcounter-Jack/JSON-Patch/releases/tag/0.5.6)

**Comment by febeling** [comment]
Confirming the fix. Thanks!
